### PR TITLE
Expose the RMM pool size in JNI

### PR DESCRIPTION
**Pull request by revans2**
## Description
This exposes the previously set pool size.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

**Review by jlowe — COMMENTED** [review]
This seems OK yet incomplete.  Wouldn't we want to get the allocation mode as well?  Seems a bit odd to expose pool size yet not even be able to tell if pooling is enabled (we don't force the pool size to be zero or -1 if pooling is not requested).

**Comment by revans2** [comment]
> This seems OK yet incomplete. Wouldn't we want to get the allocation mode as well? Seems a bit odd to expose pool size yet not even be able to tell if pooling is enabled (we don't force the pool size to be zero or -1 if pooling is not requested).

Even if pooling is disabled the pool size is used as the maximum allocation size. Should I rename the method? Add more documentation to it?

**Comment by revans2** [comment]
@jlowe please take another look

**Review by jlowe — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/Rmm.java [hunk 1]
```diff
@@ -119,6 +121,27 @@ public static synchronized void initialize(int allocationMode, LogConf logConf,
...
+  /**
+   * Get the most recently set pool size or -1 if RMM has not been initialized or pooling is
+   * not enabled.
+   */
+  public static synchronized long getPoolSize() {
+    if (isPoolingEnabled()) {
+      return poolSize;
+    } else {
+      return -1;
+    }
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/Rmm.java:137]: `poolSize` has default value `-1` right? So should we just return `poolSize`?

File: java/src/main/java/ai/rapids/cudf/Rmm.java [hunk 2]
```diff
@@ -119,6 +121,27 @@ public static synchronized void initialize(int allocationMode, LogConf logConf,
...
+  }
+
+  /**
+   * Return true if rmm is initialized and pooling has been enabled, else false.
+   */
+  public static synchronized boolean isPoolingEnabled() {
+    return initialized && poolingEnabled;
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/Rmm.java:144]:
```suggestion
    return poolingEnabled;
```
`poolingEnabled == true` only after initialization so we don't need to check for `initialized`.

File: java/src/main/java/ai/rapids/cudf/Rmm.java (hunk 1 above)
Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/Rmm.java:137]: Not really. If pooling is disabled we would return whatever the pool size is that the user passed in. But it is simple to just clean that up when we set it, so I will do that.

**Comment by revans2** [comment]
@ttnghia I think I have addressed all of your review comments.

**Review by ttnghia — APPROVED** [review]

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
@gpucibot merge

**Pull request merged**